lib/elven_gard/structures/client.ex

```diff
@@ -28,7 +28,7 @@ defmodule ElvenGard.Structures.Client do
   @doc """
   Send a packet to the client
   """
-  @spec send(__MODULE__.t(), binary) :: :ok | {:error, atom}
+  @spec send(__MODULE__.t(), term) :: :ok | {:error, atom}
```

Comment:
A socket send function should only accept binary.

Reply:
And it is the case. Except that before sending the packet, this parameter is sent to the encoder so that it can encode it. That's why I put this spec

Reply:
An encoder should have a standard entry type such as a map (like json), list (like csv) or string (like some text encoding). Like it do not make sense to have a string into a json encoder.

Reply:
You can find here an example of why I have these specs : [NosTale Encoder](https://github.com/ImNotAVirus/Flugel-NostaleEmu/blob/master/apps/login_server/lib/login_server/packet_encoder.ex#L16) and [FlyFF Encoder](https://github.com/ImNotAVirus/Imanity-FlyffEmu/blob/master/apps/login_server/lib/login_server/packet_encoder.ex#L13).  
For a Textual Protocol, I just have the packet (a string) to send to the encoder but I can't just send a simple binary for the Binary Protocol.

Btw, when you said `an encoder should have a standard input`, you're absolutely right: for the NosTale protocol, the standard is a `string` and for the FlyFF one, this is `{packet_id, data}`